### DISCUSSION TITLE
eBus signal lost

### DISCUSSION THREAD

**Discussion by alonjr** [john30/ebusd#771] [Q&A]
Hello, from time to time, eBus signal is lost (not to be confused with network connectivity with the adaptor)—see screenshot.
A couple of questions come to mind ...

1. what can cause the loss of signal?
2. how to gracefully recover and regain signal again (I find reboot as a last resort)?

My setup is ebus Wifi adaptor 3 listening to Saunier Duval THEMAFAST CONDENS 30. ebusd is running as an add-on in Home Assistant.

Happy holidays
Alon

<img width="505" alt="Screen Shot 2022-12-26 at 9 18 02 AM" src="https://user-images.githubusercontent.com/4352088/209524485-68585a28-bebb-49a4-bbb3-0e2fc9ccef0e.png">

**Comment by ulda** [comment]
Hi alonjr
I observed that  if you do not connect to the given socket , the adapter stops and sais "ebus conected: no, ebus signal: no signal" but keeps blinking and "initial signal: yes".
When I connect, either with ebusd or with a small homegrown script, it reconnects to the ebus.

Maybe you have to look , why ebus is not connected permanently ?
eg. check the log entries of the ebusd ad-in in homeassistant, see any error here?
maybe add log levels for network messages and bus messages in the config to see if it hicks via the logs ?

script to check it out in raw mode:
```
# read enhanced protocol and spit out real values

#import sys
#f=sys.stdin

import socket

s = socket.create_connection( ( "192.168.178.52",  9999 ))
f = s.makefile('rb')

ignore=False

# ask for version info
s.sendall( bytes( (0b1100_1100, 0b1000_0000 ) ) )


while True:
    try:
        c1= f.read(1)
    except KeyboardInterrupt:
        exit(0)

    if c1 == '':
        exit(0)

    i1= ord(c1)

    if i1 and 0b1000_0000 :
        c2=f.read(1)

        i2=ord(c2)

        #print('i2=%0x '% (i2 & 0b0111_1111) )

        if i2 & 0b0100_0000:
            #swap
            i1,i2 = i2,i1
            print('/',end='')

        info=( i1 & 0b0011_1100 ) >>2
        if info > 1:
            print('info=%0x '% info , end='' )
    
        o1= (i2 & 0b0011_1111) | ( (i1 & 0b0000_0011) << 6 )      
        
    else:
        o1=i1
        print('.',end='')

    # g.write(chr(o1))
    if o1 == 0xaa:
        if not  ignore:
            ignore=True
            print("")
    else:
        ignore=False
        print( "%02x "%o1, end='')   

```

bye
ulda
